subprojects/code-quality/src/test/groovy/org/gradle/api/plugins/quality/PmdPluginTest.groovy

```diff
@@ -221,6 +221,7 @@ class PmdPluginTest extends Specification {
             main
         }
         expect:
-        project.tasks.getByName("pmdMain").classpath == project.sourceSets.main.compileClasspath
+        project.tasks.getByName("pmdMain").classpath.getFiles() ==
+            project.files(project.sourceSets.main.output, project.sourceSets.main.compileClasspath).getFiles()
```

Comment:
For the sake of readability, I would suggest we simplify this a little too. Perhaps something along the lines of:

    project.tasks.getByName("pmdMain").classpath.files ==
        (project.sourceSets.main.output + project.sourceSets.main.compileClasspath).files

We could even extract some variables here for the sake of readability.